app/tests/api/route/test_user_route.py

```diff
@@ -52,216 +67,167 @@ def validate_param_match(key, request, response, db_record):
-
-
-def test_post_user_201_empty_array(client, api_auth_token, test_db_session):
-    request = base_request | {"roles": []}
-    response = client.post("/v1/user", json=request, headers={"X-Auth": api_auth_token})
-
-    assert response.status_code == 201
-
-    results = test_db_session.query(User).all()
-    assert len(results) == 1
-    db_record = results[0]
-    response_record = response.get_json()["data"]
-
-    # Verify the roles portion of the object also matches
-    request_roles = request["roles"]
-    response_roles = response_record["roles"]
-    db_roles = db_record.roles
-    assert 0 == len(request_roles) == len(response_roles) == len(db_roles)
-
-
-def test_post_user_400_missing_required_fields(client, api_auth_token, test_db_session):
-    # Send an empty post - should fail validation
-    response = client.post("/v1/user", json={}, headers={"X-Auth": api_auth_token})
-    assert response.status_code == 400
-
-    error_list = response.get_json()["detail"]["json"]
-    required_fields = [
-        "first_name",
-        "last_name",
-        "phone_number",
-        "date_of_birth",
-        "is_active",
-        "roles",
-    ]
-    assert len(error_list) == len(
-        required_fields
-    ), f"Errored fields don't match expected for empty request {error_list}"
-    for key, errors in error_list.items():
-        assert key in required_fields
-        assert "Missing data for required field." in errors
-
-    # Nothing added to DB
-    results = test_db_session.query(User).all()
-    assert len(results) == 0
-
-
-def test_post_user_400_invalid_types(client, api_auth_token, test_db_session):
-    request = {
-        "first_name": 1,
-        "middle_name": 2,
-        "last_name": 3,
-        "date_of_birth": 4,
-        "phone_number": 5,
-        "is_active": 6,
-        "roles": 7,
+    request["roles"] = roles
+    post_response = client.post("/v1/user", json=request, headers={"X-Auth": api_auth_token})
+    post_response_data = post_response.get_json()["data"]
+    expected_response = {
+        **request,
+        "id": post_response_data["id"],
+        "created_at": post_response_data["created_at"],
+        "updated_at": post_response_data["updated_at"],
     }
-    response = client.post("/v1/user", json=request, headers={"X-Auth": api_auth_token})
+
+    assert post_response.status_code == 201
+    assert post_response_data == expected_response
+    assert post_response_data["created_at"] is not None
+    assert post_response_data["updated_at"] is not None
+
+    # Get the user
+    user_id = post_response.get_json()["data"]["id"]
+    get_response = client.get(f"/v1/user/{user_id}", headers={"X-Auth": api_auth_token})
+
+    assert get_response.status_code == 200
+
+    get_response_data = get_response.get_json()["data"]
+    assert get_response_data == expected_response
+
+
+@pytest.mark.parametrize(
+    "request_data,expected_response_data",
+    [
+        pytest.param(
```

Comment:
yeah what i've done in the past is something in between, rather than a separate top level variable for every scenario, i'd do something like

```
test_data = [
    (..., ...),
    (..., ...),
    (..., ...),
]
@pytest.mark.parametrize("foo,expected", test_data)
```

i pushed up a change that does this